atom/browser/api/atom_api_web_contents.cc

```diff
@@ -119,10 +126,57 @@ struct Converter<PrintSettings> {
       return false;
     dict.Get("silent", &(out->silent));
     dict.Get("printBackground", &(out->print_background));
+    dict.Get("deviceName", &(out->device_name));
     return true;
   }
 };
 
+using StringMap = std::map<std::string, std::string>;
+
+template<>
+struct Converter<StringMap> {
+    static v8::Local<v8::Value>
+    ToV8(v8::Isolate* isolate, const StringMap& val) {
```

Comment:
There is already a FromV8 converter for `std::map<std::string, T>` in native_mate, this can go in there. https://github.com/electron/native-mate/blob/master/native_mate/converter.h#L267